build_runner/lib/src/server/hot_reload_client.dart

```diff
@@ -2,14 +2,85 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:convert';
 import 'dart:html';
+import 'dart:js';
```

Comment:
we generally prefer to use `package:js` instead of `dart:js` - can you see what it would look like doing that? It's possible not much will need to change since `package:js` exports parts of `dart:js`.

Reply:
It don't exports `JsObject` and other types I use. And using `@JS` annotation will ends up with large amount of boilerplate code

Reply:
`JsObject` is soft-deprecated:
https://github.com/dart-lang/sdk/issues/33350

We will likely be adding lints in the near future to prevent usage:
https://github.com/dart-lang/sdk/issues/33355

Using `package:js/js_util.dart` you should have access to some lower-level APIs that don't require much boiler-plate.